### PR TITLE
Restricts some more areas from servant warp, and lets admins start the Ark on non-clockcult rounds

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: Servants can no longer teleport into the gravity generator, EVA, or telecomms.
/:cl:

Added a new var to `/area` called `clockwork_warp_allowed` that, if set to false, will prevent servants from warping there. There's also `clockwork_warp_fail`, which is the fluff string shown to a servant who attempts to warp there but fails.

Also restricts the following areas:
* **Gravity Generator:** This is brainlessly easy to sabotage and it means that anyone with a slab and the ability to click can easily cripple the station. It's still easy to hack in, but no longer a cakewalk to shut down.
* **EVA:** I've been hearing of people stealing plasteel at roundstart as a servant in order to build r-walls that are very difficult to break through. I do still consider this a a legitimate strategy, but giving the servants a very fast upper hand seems bad to me, so like the gravgen, you now need to break in.
* **Telecomms:** A lot of issues with this one! Being able to warp into telecomms and permanently disable it at any time is obviously a bad thing, so this is my way to stopping that.
* **Chapel:** Thematic. It's a public-access area anyway, but this also opens up stuff like crew staging outposts in the Chapel because they know servants can't teleport in at any time and clean house.

Finally, admins can now activate the Ark by clicking on it as a ghost after three confirmation prompts. This sets a new global var, `servants_active`, to `TRUE` - meaning that the game treats the mode like clockwork cult and things like camera observers can warp you to places now. (You can also just varedit `servants_active` to 1 in order to have a clockwork cult but without the Ark.)

### CODE REVIEW

**Comment by Doppeltongue** [comment]
Is it possible to create an interaction with holy water or holy melons and the teleport? Would be thematic and would let the crew do something proactive instead of just waiting to get invaded. Not necessarily a return to chaplains blessing a water tank and then fire extinguishing down every hallway, but could be interesting to have a method to deny certain areas from being accessed by teleport.

**Comment by Xhuis** [comment]
That's very possible to do, yes

**Comment by Doppeltongue** [comment]
I guess the problem then is how to also let the cultists know what areas have been denied.

**Comment by MoreRobustThanYou** [comment]
maybe make it so holy water covered tiles require double the time to teleport

**Comment by AnonymousNow** [comment]
Is it possible to give areas a visual overlay when trying to teleport to indicate if a place is viable? Just something like a colour overlay would do. Shaded areas where the veil is too thick. Glitter for sanctified tiles.

**Comment by Xhuis** [comment]
It's likely possible, but I'm unsure how I'd do that.

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 1]
```diff
@@ -34,6 +34,16 @@
 		GLOB.ark_of_the_clockwork_justiciar = src
 	START_PROCESSING(SSprocessing, src)
 
+/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/final_countdown(ark_time)
+	if(!ark_time)
+		ark_time = 30 //minutes
+	initial_activation_delay = ark_time * 60
+	seconds_until_activation = ark_time * 60 //60 seconds in a minute * number of minutes
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: can you please make this vars, and ark_time, actual decisecond values?

**Review by ChangelingRain — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm (hunk 1 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: In order to track the countdown to the second, the Ark counts down per-second, like it also did before the rework. Making them deciseconds would have to remove a chunk of functionality from the Ark and would require a refactor.

Comment (ChangelingRain) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: world.time is deciseconds, so it's not that hard to change and keep the functionality.

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: How would I implement it? Change everything the Ark uses to use deciseconds and world.time? `process()` already ticks once per second.

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: 🔔

**Comment by JJRcop** [comment]
Note for @AnturK, remove GLOB.servants_active when killing game_mode things.

**Pull request merged**